Droidcon-Boston/app/src/main/java/com/mentalmachines/droidcon_boston/utils/DividerItemDecoration.java

```diff
@@ -0,0 +1,92 @@
+package com.mentalmachines.droidcon_boston.utils;
+
+import android.content.Context;
+import android.content.res.TypedArray;
+import android.graphics.Canvas;
+import android.graphics.Rect;
+import android.graphics.drawable.Drawable;
+import android.support.v4.view.ViewCompat;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.view.View;
+
+public class DividerItemDecoration extends RecyclerView.ItemDecoration {
+
+  private static final int[] ATTRS = new int[]{
+      android.R.attr.listDivider
+  };
+
+  public static final int HORIZONTAL_LIST = LinearLayoutManager.HORIZONTAL;
+
+  public static final int VERTICAL_LIST = LinearLayoutManager.VERTICAL;
+
+  private Drawable mDivider;
+
+  private int mOrientation;
+
+  public DividerItemDecoration(Context context, int orientation) {
+    final TypedArray a = context.obtainStyledAttributes(ATTRS);
+    mDivider = a.getDrawable(0);
+    a.recycle();
+    setOrientation(orientation);
+  }
+
+  public void drawHorizontal(Canvas c, RecyclerView parent) {
+    final int top = parent.getPaddingTop();
+    final int bottom = parent.getHeight() - parent.getPaddingBottom();
+
+    final int childCount = parent.getChildCount();
+    for (int i = 0; i < childCount; i++) {
+      final View child = parent.getChildAt(i);
+      final RecyclerView.LayoutParams params = (RecyclerView.LayoutParams) child
+          .getLayoutParams();
+      final int left = child.getRight() + params.rightMargin +
+          Math.round(ViewCompat.getTranslationX(child));
+      final int right = left + mDivider.getIntrinsicHeight();
+      mDivider.setBounds(left, top, right, bottom);
+      mDivider.draw(c);
+    }
+  }
+
+  public void drawVertical(Canvas c, RecyclerView parent) {
+    final int left = parent.getPaddingLeft();
+    final int right = parent.getWidth() - parent.getPaddingRight();
+
+    final int childCount = parent.getChildCount();
+    for (int i = 0; i < childCount; i++) {
+      final View child = parent.getChildAt(i);
+      final RecyclerView.LayoutParams params = (RecyclerView.LayoutParams) child
+          .getLayoutParams();
+      final int top = child.getBottom() + params.bottomMargin +
+          Math.round(ViewCompat.getTranslationY(child));
+      final int bottom = top + mDivider.getIntrinsicHeight();
+      mDivider.setBounds(left, top, right, bottom);
+      mDivider.draw(c);
+    }
+  }
+
```

Comment:
you can merge these methods together and use a method parameter to switch between the two.